docs/compiler-exceptions.rst

```diff
@@ -0,0 +1,287 @@
+            -----------------------^
+                 29
+
+    The terminal returns a compiling error warning.  Reading the entire warning is critical to understanding exactly what
+    is causing the error.  The message displays the line numbers that contain the error.  In this example ``map(_KeyType, _ValueType)`` cannot compile because the type ``address`` is misspelled.
+
+.. py:exception:: JSONError
+
+    Vyper has the ability to pass information back and forth using JSON.  If you are using JSON and receiving a JSON error then you can find out more details about the error at `Oracle JSON Errors <https://docs.python.org/3/tutorial/errors.html>`_.
+
+.. py:exception:: NonPayableViolationException
+
+    .. code-block:: python3
+
+        @private
+        def _foo():
+            bar: uint256 = msg.value
+       
+    Raised when attempting to access ``msg.value`` from within a private function.
+
+.. py:exception:: ParserException
+
+    .. code-block:: python3
+
+        @private
+        def foo(a: address = msg.sender):
+            pass
+
+    This function is invalid because ``msg.sender`` cannot be accessed inside of a private function.
+
+    .. code-block:: bash
+    
+        vyper.exceptions.ParserException: line 3:21 msg.sender not allowed in private functions.
+             2 @private
+        ---> 3 def foo(a: address = msg.sender): pass
+        ----------------------------^
+
+    See ``PythonSyntaxException``.  This error is not commonly used and typically refers to an error of one of the other types.
+
+.. py:exception:: PythonSyntaxException
+
+    .. code-block:: python3
+
+        >>> while True print('Hello world')
+            File "<stdin>", line 1
+            while True print('Hello world')
+                   ^
+            SyntaxError: invalid syntax
+     
+    This exception is raised due to a python based syntax exception.  It is also known as a parser exception.  In this case there is a colon missing after the word print.  See `Python Errors and Exceptions <https://docs.python.org/3/tutorial/errors.html>`_ for more details.
+
+.. py:exception:: StructureException
+
+    .. code-block:: python3
+
+        # Transfer funds to beneficiary
+        send(self.beneficiary, self.highestBid)
+
+        '''
+
+    The inclusion of punctuation that is syntactically incorrect and not a part of the normal vyper flow will throw a 
+    ``StructureException``.
+
+    .. code-block:: bash
+
+        vyper.exceptions.StructureException: line 181:0 Invalid top-level statement
+             180
+        ---> 181 '''
+        ---------^
+             182  
+    The terminal displays the line number and points directly to the problem.
+
+.. py:exception:: SyntaxException
+
+    .. code-block:: python3
+
+        struct Bid:
+            blindedBid bytes32
+            deposit: wei_value
+
+    A syntax error is thrown in the declaration of this ``struct`` variable.
+
+    .. code-block:: bash
```

Comment:
The indentation is still incorrect here. I can't do a multi-line suggestion but you can copy from below:

```
    .. code-block:: bash

        $ vyper blind_auction.vy
        vyper.exceptions.PythonSyntaxException: line 4:20 SyntaxError: invalid syntax
             3 struct Bid:
        ---> 4   blindedBid bytes32
        ---------------------------^
             5   deposit: wei_value
```

Reply:
fixed